### PR TITLE
docs: Add troubleshooting section on checkpointing for EventHub Azure Functions using the new EventHub functions extension.

### PR DESCRIPTION
**Pull request by v-shenoy**
Added a section to the EventHub scaler explaining issues with checkpointing for people using the new EventHub extension for Azure Functions.

See https://github.com/kedacore/keda/issues/2668

Signed-off-by: Vighnesh Shenoy <vshenoy@microsoft.com>

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)

### CODE REVIEW

File: content/docs/2.6/scalers/azure-event-hub.md
Comment (tomkerkhove) [inline content/docs/2.6/scalers/azure-event-hub.md:42]:
```suggestion
> 
> 💡 **Azure Functions:** Functions in C# using `Microsoft.Azure.WebJobs.Extensions.EventHubs` >= `5.0.0`, and functions in other languages that use `Microsoft.Azure.Functions.ExtensionBundle` >= `3.x` have to set `checkpointStrategy` to `blobMetadata` and `blobContainer` to `azure-webjobs-eventhub`.
```

**Review by tomkerkhove — CHANGES_REQUESTED** [review]
Can you update all scaler versions for Event Hubs with support for `checkpointStrategy` please?

**Comment by tomkerkhove** [comment]
@kedacore/keda-maintainers This warning is becoming long; maybe we should move it to a h2 or h3?

For example:
> 💡Learn more about the various checkpointing strategies in [this section]()

File: content/docs/2.6/scalers/azure-event-hub.md
Comment (v-shenoy) [reply content/docs/2.6/scalers/azure-event-hub.md:42]: Will do. I'll wait on the decision about moving these to a separate section, before updating the PR.

**Comment by JorTurFer** [comment]
> @kedacore/keda-maintainers This warning is becoming long; maybe we should move it to a h2 or h3?
> 
> For example:
> 
> > 💡Learn more about the various checkpointing strategies in this section

+1 for another section

File: content/docs/2.3/scalers/azure-event-hub.md
Comment (tomkerkhove) [inline content/docs/2.3/scalers/azure-event-hub.md:63]: Three things I would suggest is:
1. Add link to the values that you can specify in `checkpointStrategy` as we don't talk about legacy & current there
2. Move the Frameworks and respective packages to a list so that it's easier to find, maybe? (example below)
3. Cover remaining strategies (OK in separate PR if you prefer to)

```suggestion
* **Legacy behaviour:** Checkpoint information is stored as blob container. Default behavior when no `checkpointStrategy` is specified.
  - .NET applications using `Microsoft.Azure.EventHubs` NuGet package
  - Java  applications using `azure-eventhubs-eph` {?} package
  - Python applications using `azure-eventhub` {?} package below 5.0
```

Comment (v-shenoy) [reply content/docs/2.3/scalers/azure-event-hub.md:63]: Could you be a bit more specific on what points should be covered for the remaining strategies?

Comment (tomkerkhove) [reply content/docs/2.3/scalers/azure-event-hub.md]: With 3) I was more referring to this:
image

Where we currently talk about legacy and current, but what about the other options for `checkpointStrategy` but maybe I'm overthikning things @kedacore/keda-maintainers?

Comment (v-shenoy) [reply content/docs/2.3/scalers/azure-event-hub.md]: I am not yet sure on what I should be adding besides the information already mentioned under the parameters section. Happy to raise a separate PR for it, if someone clarifies that to me.

Comment (tomkerkhove) [reply content/docs/2.3/scalers/azure-event-hub.md]: I've had another look and let's just leave it as-is, we can still improve if need be.

File: content/docs/2.3/scalers/azure-event-hub.md
Comment (tomkerkhove) [inline content/docs/2.3/scalers/azure-event-hub.md:68]: This might be better?
```suggestion
This is applicable for the following scenarios:
```

File: content/docs/2.3/scalers/azure-event-hub.md
Comment (tomkerkhove) [inline content/docs/2.3/scalers/azure-event-hub.md:75]: This might be better?
```suggestion
This is applicable for the following scenarios:
```

**Review by tomkerkhove — APPROVED** [review]

File: content/docs/2.3/scalers/azure-event-hub.md
Comment (v-shenoy) [reply content/docs/2.3/scalers/azure-event-hub.md]: Done.

**Comment by tomkerkhove** [comment]
Thanks a ton and sorry for the delay!

**Pull request merged**
